lib/spack/spack/package.py

```diff
@@ -2048,13 +2048,9 @@ def _run_default_build_time_test_callbacks(self):
             return
 
         for name in self.build_time_test_callbacks:
-            try:
```

Comment:
This try-except was masking a bug in my code that I didn't discover until after I removed it. Personally, I think we shouldn't both catching the exception, and raise an error if a non-existent test function is requested. For most build systems, we implement an empty test method with `pass` anyway, so it shouldn't fail.